indexer/services/roundtable/__tests__/tasks/update-affiliate-info.test.ts

```diff
@@ -0,0 +1,245 @@
+  AffiliateInfoFromDatabase,
+  AffiliateInfoTable,
+  Liquidity,
+} from '@dydxprotocol-indexer/postgres';
+import affiliateInfoUpdateTask from '../../src/tasks/update-affiliate-info';
+import { DateTime } from 'luxon';
+
+describe('update-affiliate-info', () => {
+  beforeAll(async () => {
+    await dbHelpers.migrate();
+    await dbHelpers.clearData();
+  });
+
+  beforeEach(async () => {
+    await testMocks.seedData();
+    await OrderTable.create(testConstants.defaultOrder);
+  });
+
+  afterAll(async () => {
+    await dbHelpers.teardown();
+    jest.resetAllMocks();
+  });
+
+  afterEach(async () => {
+    await dbHelpers.clearData();
+  });
+
+  it('Successfully updates affiliate info and persistent cache multiple times', async () => {
+    const startDt = DateTime.utc();
+
+    // Set persistent cache affiliateInfoUpdateTIme so task does not use backfill windows
+    await PersistentCacheTable.create({
+      key: PersistentCacheKeys.AFFILIATE_INFO_UPDATE_TIME,
+      value: startDt.toISO(),
+    });
+
+    // First task run: add refereal w/o any fills
+    // defaultWallet2 will be affiliate and defaultWallet will be referee
+    await AffiliateReferredUsersTable.create({
+      affiliateAddress: testConstants.defaultWallet2.address,
+      refereeAddress: testConstants.defaultWallet.address,
+      referredAtBlock: '1',
+    });
+    await affiliateInfoUpdateTask();
+
+    let updatedInfo: AffiliateInfoFromDatabase | undefined = await AffiliateInfoTable.findById(
+      testConstants.defaultWallet2.address,
+    );
+    let expectedAffiliateInfo: AffiliateInfoFromDatabase = {
+      address: testConstants.defaultWallet2.address,
+      affiliateEarnings: '0',
+      referredMakerTrades: 0,
+      referredTakerTrades: 0,
+      totalReferredFees: '0',
+      totalReferredUsers: 1,
+      referredNetProtocolEarnings: '0',
+      firstReferralBlockHeight: '1',
+      referredTotalVolume: '0',
+    };
+    expect(updatedInfo).toEqual(expectedAffiliateInfo);
+
+    // Check that persistent cache updated
+    const lastUpdateTime1 = await getAffiliateInfoUpdateTime();
+    if (lastUpdateTime1 !== undefined) {
+      expect(lastUpdateTime1.toMillis())
+        .toBeGreaterThan(startDt.toMillis());
+    }
+
+    // Second task run: one new fill and one new referral
+    await FillTable.create({
+      ...testConstants.defaultFill,
+      liquidity: Liquidity.TAKER,
+      createdAt: DateTime.utc().toISO(),
+      eventId: testConstants.defaultTendermintEventId,
+      price: '1',
+      size: '1',
+      fee: '1000',
+      affiliateRevShare: '500',
+    });
+    await AffiliateReferredUsersTable.create({
+      affiliateAddress: testConstants.defaultWallet2.address,
+      refereeAddress: testConstants.defaultWallet3.address,
+      referredAtBlock: '2',
+    });
+
+    await affiliateInfoUpdateTask();
+
+    updatedInfo = await AffiliateInfoTable.findById(
+      testConstants.defaultWallet2.address,
+    );
+    expectedAffiliateInfo = {
```

Comment:
better to use a second variable, and keep updatedInfo and expectedAffiliateInfo as `const`